### PR TITLE
feat: ens name display support

### PR DESCRIPTION
**Pull request by Rabursky**
Screenshot 2021-06-11 at 12 07 19

It was frustrating not to support ENS names...

Thanks for having this website open sourced ✌️

### CODE REVIEW

File: src/components/parts/ENSName/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const [ name, setName ] = useState<string>()
+    useEffect(() => {
+        getNameForAddress(address, provider).then(({ name }) => setName(name))
+    })
+    return name
+}
+
+export default function ENSName({ address, fallback = null, provider = window.web3 ? window.web3.currentProvider : window.ethereum }: { address: string, fallback?: string, provider?: any }) {
+    const name = useENSName(address, provider)
+    return <>{name || fallback}</>
```
Comment (rootulp) [inline src/components/parts/ENSName/index.tsx:19]: This looks like an ideal usecase for https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
```suggestion
    return <>{name ?? fallback}</>
```

**Review by rootulp — APPROVED** [review]
Thanks for your contribution!

**Review by wpmonty — APPROVED** [review]
lgtm; @nateawelch any opinions?

File: src/components/parts/ENSName/index.tsx (hunk 1 above)
Comment (Rabursky) [reply src/components/parts/ENSName/index.tsx:19]: I think I would prefer to go with OR in this case. I'd rather fallback to truncated ETH address if ENS returns something falsy like `0` or `''`. Better safe than sorry 🤷‍♂️

File: src/components/parts/ENSName/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  
+function useENSName(address, provider) {
+    const [ name, setName ] = useState<string>()
+    useEffect(() => {
+        getNameForAddress(address, provider).then(({ name }) => setName(name))
+    })
+    return name
+}
+
+export default function ENSName({ address, fallback = null, provider = window.web3 ? window.web3.currentProvider : window.ethereum }: { address: string, fallback?: string, provider?: any }) {
```
Comment (Oktalize) [inline src/components/parts/ENSName/index.tsx:17]: I found an error when using Wallet Connect. The provider doesn't get passed into to the `ENSName` component and it triggers a runtime error when it is missing.

One possible solution would be to pass the provider from the `web3Modal` in `ConnectWalletButton.tsx` which would avoid the use of `window` here. I can get this fixed for you if you need help.

CleanShot 2021-06-16 at 11 55 24

**Review by Oktalize — CHANGES_REQUESTED** [review]

**Review by Oktalize — APPROVED** [review]
I have no remaining issues now that wallet connect is fixed. I tested the fix in different browsers and wallets and it looks good.

**Pull request merged**